events/controller/status.go

```diff
@@ -186,7 +186,7 @@ func (s *NatsConditionStatusPublisher) update(key string, newStatusValue *condit
 	}
 
 	if curStatusValue.WorkerID != s.controllerID {
-		return 0, errors.Wrap(errWorkerMismatch, curStatusValue.WorkerID)
+		return 0, errors.Wrap(errControllerMismatch, curStatusValue.WorkerID)
```

Comment:
This feels like the wrong place to catch this error, but thinking about it this is not the case where a controller is starting work inappropriately but rather where a tardy, partitioned worker is trying to update status when another controller has been assigned.
1. a comment to that effect might be instructive.
2. This is a messy situation and I don't know that there is a good way to deal with it in code.

Reply:
ack, I shall follow up on this in the upcoming PR